### PR TITLE
Scale node labels with the node's size.

### LINKED ISSUE #1692: increasing vertical window size decreases node label font size
Showing exactly the same topology, at exactly the same zoom level / node size / node space, but with different vertical window size, changes the font size on node labels - the taller the window the smaller (paradoxically) the font size gets.

The following results were obtained by opening fresh browser windows, sizing them, and then entering the local scope url.

short window
screenshot_2016-07-17_15-58-03
tall window
screenshot from 2016-07-17 15-58-29

Happens in both Chrome and FF.

### LINKED ISSUE #1763: Details panel does not spawn at node location on click
This used to work before #1756. Now it spawns the below the screen (seen in Chrome and Firefox).
Weirdly, the origin's x coordinate still works.

### PR DESCRIPTION
**Pull request by foot**
- Within certain bounds. Still have min-label size, mainly effects nodes
  that get really big.
- Set a max size on nodes too, really big ones lose their border.

Fixes #1692 
Fixes #1763

### CODE REVIEW

**Comment by foot** [comment]
Nodes prior to max size restriction

localhost-4042-

**Comment by foot** [comment]
Max size:

localhost-4042- 1

**Comment by foot** [comment]
That is for the base zoom level.

Zooming you can decrease/increase still.

**Comment by foot** [comment]
Hint: chrome devtools device-toolbar is handy for testing window sizes.

**Comment by rade** [comment]
This does appear to work. I have no idea how. Hopefully @davkal does.

**Comment by davkal** [comment]
This fix prevents the bad label overlaps, great!

Two visual issues:

1. select on node, look at label size, deselect, zoom out, select same node, label size is now bigger

2. when selecting a node, it shortly blows up in size, then the size converges to its end state

**Comment by davkal** [comment]
Code looks good.
At some point we should implement a proper semantic zoom, that spaces out the nodes a bit more and allows labels to expand.

**Comment by foot** [comment]
screen shot 2016-08-09 at 18 25 20

Looks kinda fun

**Comment by foot** [comment]
Ready for re-review

**Comment by rade** [comment]
So does this now also fix #1763?

**Comment by foot** [comment]
Yep, issue updated.

File: client/app/scripts/charts/node.js [hunk 1]
```diff
@@ -126,9 +125,9 @@ class Node extends React.Component {
           svgLabels(label, subLabel, labelClassName, subLabelClassName, labelOffsetY) :
 
           <foreignObject style={{pointerEvents: 'none'}} x={labelOffsetX} y={labelOffsetY}
-            width={labelWidth} height="100em" transform={labelTransform}>
+            width={labelWidth} height="100em">
             <div className="node-label-wrapper"
-              style={{pointerEvents: 'all', fontSize, maxWidth: labelWidth}}
+              style={{pointerEvents: 'all', fontSize, lineHeight: '150%', maxWidth: labelWidth}}
```
Comment (davkal) [inline client/app/scripts/charts/node.js]: Can we put static numbers into the .less file?

File: client/app/styles/main.less [hunk 2]
```diff
@@ -546,7 +546,7 @@ h2 {
   text-align: center;
 
   &-match {
-    font-size: 0.8rem;
+    font-size: 0.8em;
```
Comment (davkal) [inline client/app/styles/main.less]: Why not rem?

File: client/app/scripts/charts/node.js (hunk 1 above)
Comment (foot) [inline client/app/scripts/charts/node.js]: Yes, this didn't hop across the foreignObject properly I think I can put in the less file though.

**Comment by davkal** [comment]
LGTM

**Comment by foot** [comment]
This issue fixes label truncation (ellipsis) too, but I don't think thats the same issue as #1760

**Comment by rade** [comment]
> This issue fixes label truncation (ellipsis) too

What exactly is it fixing? Have you got a before/after screenshot?

**Comment by foot** [comment]
screen shot 2016-08-10 at 14 17 37
screen shot 2016-08-10 at 14 17 27

**Comment by rade** [comment]
oh ok. I hadn't actually noticed that. What does one have to do to provoke this? I've played with zoom and +/- to no avail.

**Comment by foot** [comment]
Nothing in particular just have nice long container names, its noticeable for me on proud-wind.

**Comment by rade** [comment]
got it. needed to shrink my window vertically to see the problem.

**Pull request merged**